oauth_adapter.py

```diff
@@ -21,6 +22,21 @@ def __init__(self, client_id, client_secret, open_id_config_url, provider_name):
         self.basic_auth = HTTPBasicAuth(self.client_id, self.client_secret)
         self.provider_name = provider_name
 
+    def build_authz_url(self, scopes, redirect_uri, state=None):
+        """
+        Builds an OAuth authorization URL that a user must use to initiate the OAuth dance.
+        :param scopes: Array of scopes (0 to many) that the client requires
+        :param redirect_uri: A URL encoded string representing the URI that the Authorizing Service will redirect the
+        user to after the user successfully authorizes this client
+        :param state: A URL encoded Base64 string representing a JSON object of state information that the requester
+        requires back with the redirect
+        :return: A plain (not URL encoded) String
+        """
+        fence_authorize_url = requests.get(self.open_id_config_url).json()["authorization_endpoint"]
```

Comment:
this is not fence specific so don't call the var fench

also use self. _get_open_id_config to get the openid config instead of requesting it each time